src/main/groovy/wooga/gradle/build/unity/UnityBuildPlugin.groovy

```diff
@@ -220,7 +217,19 @@ class UnityBuildPlugin implements Plugin<Project> {
                         project.tasks.getByName(taskName).dependsOn(gradleBuild)
                     }
                 }
+
             }
         }
     }
+
+    static void configureSonarqubeTasks(Project project) {
+        def unityExt = project.extensions.findByType(UnityPluginExtension)
+        def sonarExt = project.extensions.findByType(SonarQubeExtension)
+
+        new SonarQubeConfiguration(project).with {
+            sonarTaskName = "sonarqube"
+            buildTaskName = "sonarBuildUnity"
+            return it
+        }.configure(unityExt, sonarExt)
```

Comment:
Where does the `SonarQubeConfiguratin` come from? We never used such a setup for configuration before. Just so I know where I can lookup the logic.

Reply:
Its a separated class. I did it like this because the sonarqube configuration was starting to get too big to just leave it on the plugin class.